csharp/src/Apache.Arrow/Memory/LargeMemory.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Runtime.InteropServices;
+using System.Buffers;
+using System.Runtime.CompilerServices;
+
+namespace Apache.Arrow.Memory
+{
+    internal class LargeMemory<T>: IDisposable
+    {
+        private IntPtr _ptr;
+        private long _size;
+
+        public LargeMemory(long size)
```

Comment:
I think we should try to make these types match the standard, non-large types as closely as possible. Eg. you can see the source for `System.Memory` [here](https://github.com/dotnet/runtime/blob/5535e31a712343a63f5d7d796cd874e563e5ac14/src/libraries/System.Private.CoreLib/src/System/Memory.cs). In .NET the `Memory<T>` type doesn't have a constructor that will allocate memory, and it isn't disposable. It's a struct that holds a reference to an array or a `MemoryManager`. I think for simplicity we can skip supporting an array backed `LargeMemory` and only work with a `LargeMemoryManager`, so `LargeMemory` should be quite simple and mostly delegate functionality to a `LargeMemoryManager`.

The `MemoryManager` is the thing that is disposable and can actually manage the memory, but this is an abstract class so it doesn't implement memory allocation or disposal directly. In Arrow there's a [NativeMemoryManager](https://github.com/apache/arrow/blob/main/csharp/src/Apache.Arrow/Memory/NativeMemoryManager.cs), which inherits from `MemoryManager`, and I think a lot of the functionality currently in this `LargeMemory` class should be in a new `LargeNativeMemoryManager` class instead, which should look pretty similar to the `NativeMemoryManager`.

Arrow additionally separates the functionality for allocating memory from the `MemoryManager`, eg. with the [NativeMemoryAllocator](https://github.com/apache/arrow/blob/main/csharp/src/Apache.Arrow/Memory/NativeMemoryAllocator.cs) class.

In order to integrate this new large memory functionality with Arrow, we might want to do something like extend the `MemoryAllocator` abstract base class by adding a new method, `public ILargeMemoryOwner<byte> AllocateLarge(long length)`. Then the existing `NativeMemoryAllocator` could be extended to create either a `NativeMemoryManager` or `LargeNativeMemoryManager` depending on which method is used.